src/status_im/chat/events.cljs

```diff
@@ -75,14 +83,41 @@
   (fn [db [{:keys [chat-id message-id]}]]
     (update-in db [:chats chat-id :messages message-id] assoc :appearing? false)))
 
+(handlers/register-handler-fx
+  :set-message-envelope-hash
+  [re-frame/trim-v]
+  (fn [{:keys [db]} [chat-id message-id envelope-hash]]
+    {:db (assoc-in db [:message-envelopes envelope-hash] {:chat-id    chat-id
+                                                          :message-id message-id})}))
+
+(handlers/register-handler-fx
+  :signals/envelope-status
+  [re-frame/trim-v]
+  (fn [{:keys [db] :as cofx} [envelope-hash status]]
+    (let [{:keys [chat-id message-id]} (get-in db [:message-envelopes envelope-hash])
+          message (get-in db [:chats chat-id :messages message-id])]
+      (models.message/update-message-status message status cofx))))
+
 (handlers/register-handler-fx
   :update-message-status
   [re-frame/trim-v]
-  (fn [{:keys [db]} [chat-id message-id user-id status]]
-    (let [msg-path [:chats chat-id :messages message-id]
-          new-db   (update-in db (conj msg-path :user-statuses) assoc user-id status)]
-      {:db                        new-db
-       :data-store/update-message (-> (get-in new-db msg-path) (select-keys [:message-id :user-statuses]))})))
+  (fn [{:keys [db] :as cofx} [chat-id message-id status]]
+    (let [message (get-in db [:chats chat-id :messages message-id])]
+      (models.message/update-message-status message status cofx))))
+
+;; Change status of messages which are still in "sending" status to "not-sent"
+;; (If signal from status-go has not been received)
+(handlers/register-handler-fx
+  :process-pending-messages
+  [re-frame/trim-v]
+  (fn [{:keys [db]} []]
+    (let [pending-messages (->> db
+                                :chats
+                                vals
+                                (mapcat (comp vals :messages))
+                                (filter (fn [{:keys [from user-statuses]}] (= :sending (get user-statuses from)))))]
+      (doseq [{:keys [chat-id message-id]} pending-messages]
+        (re-frame/dispatch [:update-message-status chat-id message-id :not-sent])))))
```

Comment:
I believe we can do this without dispatching a lot of (one for each pending message) unnecessary events. 
Is `:update-message-status` dispatched in any other place then `:process-pending-messages` ? If not, it should be just plain cofx taking, fx returns function and we should compose them to produce one effect where everything is encapsulated.

Reply:
@janherich good point, fixed.